### PR TITLE
[HOPSWORKS-2695] Replace creator with UserDTO in FeaturestoreEntityDTO

### PR DESCRIPTION
**Pull request by bubriks**

### CODE REVIEW

File: java/src/main/java/com/logicalclocks/hsfs/metadata/User.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+package com.logicalclocks.hsfs.metadata;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+public class User {
```
Comment (SirOibaf) [inline java/src/main/java/com/logicalclocks/hsfs/metadata/User.java:24]: You don't need to add all the fields here. First the backend doesn't populate most of them. Second you can add `@JsonIgnoreProperties(ignoreUnknown = true)` as class annotation to not make it fail in the case we ever change the backend.

File: python/hsfs/feature_group.py [hunk 2]
```diff
@@ -468,7 +468,7 @@ def __init__(
         self._feature_store_name = featurestore_name
         self._description = description
         self._created = created
-        self._creator = creator
+        self._creator = user.User.from_response_json(creator)
```
Comment (SirOibaf) [inline python/hsfs/feature_group.py:471]: Did you test this? the `creator` field is available only if the data comes from the backend. If the class is initialized by the user through the API `creator` is going to be null.

If you see the code below the one you changed, you see that we differentiate between the case in which the class is instantiated with data coming from the backend or data added by the user.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

File: python/hsfs/feature_group.py (hunk 2 above)
Comment (bubriks) [reply python/hsfs/feature_group.py:471]: if creator is null, then result of user.User.from_response_json(creator) is also None

Comment (bubriks) [reply python/hsfs/feature_group.py:471]: here is an example:

Comment (bubriks) [reply python/hsfs/feature_group.py:471]: image

Comment (bubriks) [reply python/hsfs/feature_group.py:471]: i could change it as follows, if it is still needed:
if creator is not None:
    self._creator = user.User.from_response_json(creator)
else:
    self._creator = None

**Review by SirOibaf — APPROVED** [review]

**Pull request merged**